source/io/module_eggShell.js

```diff
@@ -239,6 +246,257 @@
+        };
+
+        Module.eggShell.typeIsPath = function (typePointer) {
+            return Module._TypeRef_IsPath(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsTimestamp = function (typePointer) {
+            return Module._TypeRef_IsTimestamp(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsComplex = function (typePointer) {
+            return Module._TypeRef_IsComplex(typePointer) !== 0;
+        };
+
+        Module.eggShell.typeIsAnalogWaveform = function (typePointer) {
+            return Module._TypeRef_IsAnalogWaveform(typePointer) !== 0;
+        };
+
+        var validateVisitMethod = function (fn, fnName) {
+            if (typeof fn !== 'function') {
+                throw new Error('Visitor must have a method named `' + fnName + '`. Found: ' + fn);
+            }
+        };
+
+        var dispatchVisitBoolean = function (typeVisitor, args) {
+            validateVisitMethod(typeVisitor.visitBoolean, 'visitBoolean');
+            return typeVisitor.visitBoolean.apply(typeVisitor, args);
+        };
+
+        var dispatchVisitEnum = function (typeVisitor, args) {
+            validateVisitMethod(typeVisitor.visitEnum, 'visitEnum');
+            return typeVisitor.visitEnum.apply(typeVisitor, args);
+        };
+
+        var dispatchVisitInteger = function (typeVisitor, args) {
+            var typeRef = args[0].typeRef;
+            var isSignedInteger = Module.eggShell.typeIsSigned(typeRef);
+            var sizeOfInteger = Module.eggShell.typeTopAQSize(typeRef);
+            var visitFn = undefined;
+            if (isSignedInteger === true) {
+                switch (sizeOfInteger) {
+                case 1:
+                    visitFn = typeVisitor.visitInt8;
+                    break;
+                case 2:
+                    visitFn = typeVisitor.visitInt16;
+                    break;
+                case 4:
+                    visitFn = typeVisitor.visitInt32;
+                    break;
+                case 8:
+                    visitFn = typeVisitor.visitInt64;
+                    break;
+                default:
+                    throw new Error('Unexpected size for Integer');
+                }
+            } else {
+                switch (sizeOfInteger) {
+                case 1:
+                    visitFn = typeVisitor.visitUInt8;
+                    break;
+                case 2:
+                    visitFn = typeVisitor.visitUInt16;
+                    break;
+                case 4:
+                    visitFn = typeVisitor.visitUInt32;
+                    break;
+                case 8:
+                    visitFn = typeVisitor.visitUInt64;
+                    break;
+                default:
+                    throw new Error('Unexpected size for Unsigned Integer. Found: ');
+                }
+            }
+
+            var intName = (isSignedInteger ? '' : 'U') + 'Int' + (sizeOfInteger * 8);
```

Comment:
Can we have the string constants assigned in the individual cases instead of string building every time? Seems like extra work for what is used for error messages.

Reply:
Agree, I'll change it.